MapboxNavigation/NavigationOptions.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+import MapboxCoreNavigation
+
+/*
+    Object representing available options for a Navigation Session.
+ */
+
+@objc(MBNavigationOptions)
+open class NavigationOptions: NSObject {
+    
+    /*
+     The styles that the view controller’s internal `StyleManager` object can select from for display. Defaults to `[DayStyle(), NightStyle()]`
+     */
+    @objc public var styles: [Style]? = nil
+    
+    /*
+     The navigation service that manages navigation along the route.
+     */
+    @objc open var navigationService: NavigationService?
+    
+    /*
+     The voice controller that manages the delivery of voice instructions during navigation.
+    */
+    @objc public var voiceController: RouteVoiceController?
+    
+    /*
+     The view controlelr to embed into the bottom section of the UI. Defaults to `BottomBannerViewController`.
+     */
+    @objc public var bottomBanner: ContainerViewController?
+    
+    @objc public convenience init(styles: [Style]? = nil, navigationService: NavigationService? = nil, voiceController: RouteVoiceController? = nil, bottomBanner: ContainerViewController? = nil) {
```

Comment:
We should draw a line somewhere and omit nonessential arguments from this initializer in favor of the properties. It isn’t a goal to allow the developer to specify all the options in a single statement. `navigationService` and `styles` seem more commonly overridden than the others, but I would even be comfortable with having no convenience initializer (just the one factory initializer for Objective-C). The test I’d use is: if the developer isn’t specifically looking to customize some aspect of the experience, would forgetting its property lead to unexpected behavior, or would the default be intuitive?

Reply:
The idea here is to have struct-like initialization semantics, even though the type is not a struct. Since every initialization argument is optional with a default value of `nil`, an omitted argument in the initializer (just like an unset property), means "use default behavior."